noark-extraction-validator/src/main/java/com/documaster/validator/validation/noark53/Noark53Validator.java

```diff
@@ -239,23 +236,26 @@ private void storeXMLFiles() throws Exception {
 
 		LOGGER.info("Storing XML data and extracting document information ...");
 
-		for (String xmlFilename : getCommand().getProperties().getPackageStructure().getAllXMLFiles()) {
+		for (Noark53PackageEntity entity : structure.values()) {
 
-			File xmlFile = new File(getCommand().getExtractionDirectory(), xmlFilename);
+			if (!entity.getXmlFile().isFile() && entity.getIsOptional()) {
+				// Ignore optional entities if not found
+				continue;
```

Comment:
I'd add a log message here (possibly INFO) and in all other places where we're skipping optional entities that don't exist.